### PR TITLE
89 parametrize figure generation

### LINKED ISSUE #89: Parametrize figure generation
- [ ] Describe desired data to show  on benchmark config json
- [ ] Describe plot types to show
- [ ] Update the workflow to memorise this input across benchmarks (maybe store asciidoc reports) and make the website persistent.

### PR DESCRIPTION
**Pull request by JavierCladellas**
- Closes #89 

Figure generation is now dynamic, and is specified by the `plots` field of the config json. 
This plots config is uploaded inside the same item as the generated reframe report to girder.

example of plot configuration : 
```json 
[
        {
            "title": "Plot title",
            "plot_types":[ "scatter","table"], // Supported types atm : "scatter", "table", "stacked_bar"
            "transformation":"performance", //Supported transformations: "performance", "relative_performance", "speedup"
            "variables": [
                "ElectricConstructor_createMesh",
                "ElectricConstructor_createExporters",
                "ElectricConstructor_graph"
            ],
            "names": [
                "createMesh",
                "createExporters",
                "graph"
            ],
            "xaxis": {
                "parameter": "nb_tasks",
                "label": "Number of tasks"
            },
            "secondary_axis": {
                "parameter": "hsize",
                "label": "Mesh size"
            },
            "yaxis": {
                "label": "execution time (s)"
            }
        }
]
```

### CODE REVIEW

**Comment by JavierCladellas** [comment]
Netlify preview does not work because it is not familiar with the package. @prudhomm Do you have any solution in mind for this ? Otherwise, we could, as a hotfix, we could test if the package can be found and use relative imports if not.

**Comment by JavierCladellas** [comment]
The code is now compatible with Python 3.8 for Netlify, the preview is up @prudhomm .
Some figures ( containing a slider axis) do not appear yet. This will be fixed with the release for https://github.com/feelpp/feelpp-asciidoctor-extensions/issues/17

**Comment by prudhomm** [comment]
it is great!!!

**Comment by prudhomm** [comment]
I have to go to my course, let's discuss this afternoob

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -42,10 +42,50 @@ jobs:
...
 
+  generate_reports:
+    runs-on: self-ubuntu-22.04
+    needs: build_wheel
+    if: github.event_name == 'pull_request'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          lfs: true
+          submodules: 'recursive'
```
Comment (prudhomm) [inline .github/workflows/ci.yml:53]: are there submodules ?

File: reports/feelpp_toolbox_electric/gaya/2024_10_09T14_26_38/plots.json [hunk 2]
```diff
@@ -0,0 +1,178 @@
+[
+    {
+        "title": "ElectricConstrucor step absolute performance",
```
Comment (prudhomm) [inline reports/feelpp_toolbox_electric/gaya/2024_10_09T14_26_38/plots.json:3]: typo

**Review by prudhomm — APPROVED** [review]

**Pull request merged**